com.unity.multiplayer.mlapi/Prototyping/NetworkTransform.cs

```diff
@@ -11,364 +11,217 @@ namespace MLAPI.Prototyping
     [AddComponentMenu("MLAPI/NetworkTransform")]
     public class NetworkTransform : NetworkBehaviour
     {
-        internal class ClientSendInfo
+        public enum Authority
         {
-            public float LastSent;
-            public Vector3? LastMissedPosition;
-            public Quaternion? LastMissedRotation;
+            Server = 0, // default
+            Client,
+            Shared
         }
 
+        [SerializeField, Tooltip("Defines who can update this transform.")]
+        public Authority authority; // todo Luke mentioned an incoming system to manage this at the NetworkBehaviour level, lets sync on this
```

Comment:
I don't like public fields but ok 😃

Reply:
To elaborate. This should not be changed during runtime by code. Because it would have to be changed on every machine at the same time. So probably should be private. (but public fields in general are evil as well 😄 )

Reply:
If I want to programmatically spawn NetworkTransforms, I for sure want to update their settings at creation time.
The more I'm thinking about it, the more I think I should go with a property. It'll make future refactors easier and less risk of breaking the API for users.

Reply:
Actually scratch that. It'll just add noise for nothing right now (I need to add FormerlySerializedAs everywhere if we want to keep some form of compatibility with previous versions). If we ever want to update those vars to properties later, I'd say let's then add FormerlySerializedAs and not worry about those for now.